From 0a4db4efe8076b61967dd56ffc1745232047d121 Mon Sep 17 00:00:00 2001
From: "guoping.li" <guoping.li@amlogic.com>
Date: Fri, 10 Nov 2023 14:15:16 +0800
Subject: [PATCH 02/14] Multimedia: seperate build output files for gst aml
 utils [1/2]

PD#SWPL-145541

Problem:
build output file *.o/*.so/bin is in the same file as the source code
it cause git error

Solution:
change the makefile and bb, set build output file to build/temp/work...

Verify:
local

Signed-off-by: guoping.li <guoping.li@amlogic.com>
Change-Id: I5e87eaebd49629e386190dbc73ebb2d206e3208a
---
 conf/machine/mesonsc2-ah212-weston-5.15.config       |  1 +
 recipes-multimedia/gst-aml-utils/detect-library.bb   | 10 ++++------
 recipes-multimedia/gst-aml-utils/dma-allocator.bb    | 12 +++++-------
 .../gst-aml-utils/gst-aml-dma-allocator.bb           | 11 +++++------
 recipes-multimedia/gst-aml-utils/gst-aml-gfx2d.bb    | 11 ++++-------
 recipes-multimedia/gst-aml-utils/gst-jpeg-swdec.bb   | 10 ++++------
 recipes-multimedia/gst-aml-utils/gst-plugin-conv.bb  |  7 +++----
 recipes-multimedia/gst-aml-utils/gst-plugin-nn.bb    |  7 +++----
 .../gst-aml-utils/gst-plugin-overlay.bb              |  7 +++----
 9 files changed, 32 insertions(+), 44 deletions(-)

diff --git a/conf/machine/mesonsc2-ah212-weston-5.15.config b/conf/machine/mesonsc2-ah212-weston-5.15.config
index 5ed79d46..5aa94088 100644
--- a/conf/machine/mesonsc2-ah212-weston-5.15.config
+++ b/conf/machine/mesonsc2-ah212-weston-5.15.config
@@ -5,6 +5,7 @@
 
 include conf/machine/include/mesonsc2_k5.15_64b.inc
 include conf/machine/include/amlogic_weston.inc
+include conf/machine/include/amlogic_aiot_common.inc
 
 PREFERRED_PROVIDER_virtual/mesa = "libgles-eabihf-dvalin-wayland-drm"
 PREFERRED_PROVIDER_virtual/mesa-gl = "libgles-eabihf-dvalin-wayland-drm"
diff --git a/recipes-multimedia/gst-aml-utils/detect-library.bb b/recipes-multimedia/gst-aml-utils/detect-library.bb
index a18837aa..560a8298 100755
--- a/recipes-multimedia/gst-aml-utils/detect-library.bb
+++ b/recipes-multimedia/gst-aml-utils/detect-library.bb
@@ -13,18 +13,18 @@ SRCREV ?="${AUTOREV}"
 
 do_configure[noexec] = "1"
 
-EXTRA_OEMAKE = "TARGET_DIR=${D} STAGING_DIR=${STAGING_DIR_TARGET} DESTDIR=${D} PKG_CONFIG=${STAGING_BINDIR_NATIVE}/pkg-config"
+EXTRA_OEMAKE = "OUT_DIR=${B} TARGET_DIR=${D} STAGING_DIR=${STAGING_DIR_TARGET} DESTDIR=${D} PKG_CONFIG=${STAGING_BINDIR_NATIVE}/pkg-config"
 
 do_compile(){
     oe_runmake -C ${S} ${EXTRA_OEMAKE} all
 }
 
 do_install() {
-    install -d -m 0755 ${D}${libdir}
     install -d -m 0755 ${D}${includedir}
+    install -d -m 0755 ${D}${libdir}
 
-    install -D -m 0755 ${S}/*.so ${D}${libdir}/
-    install -D -m 0755 ${S}/inc/*.h ${D}${includedir}/
+    install -D -m 0644 ${S}/inc/*.h ${D}${includedir}/
+    install -D -m 0644 ${B}/*.so ${D}${libdir}/
 
     if [ "${HOST_ARCH}" = "aarch64" ]; then
         install -m 0644 -D ${S}/lib/lib64/*.so ${D}/usr/lib
@@ -33,8 +33,6 @@ do_install() {
     mkdir -p ${D}/etc/nn_data
     cp -rf ${S}/nn_data ${D}/etc/
     chmod -R 777 ${D}/etc/nn_data
-
-    rm ${S}/*.so
 }
 
 FILES_${PN} = " ${libdir}/* /etc/* "
diff --git a/recipes-multimedia/gst-aml-utils/dma-allocator.bb b/recipes-multimedia/gst-aml-utils/dma-allocator.bb
index e5e8a14f..106f2ff9 100755
--- a/recipes-multimedia/gst-aml-utils/dma-allocator.bb
+++ b/recipes-multimedia/gst-aml-utils/dma-allocator.bb
@@ -15,20 +15,18 @@ SRCREV ?="${AUTOREV}"
 
 do_configure[noexec] = "1"
 
-EXTRA_OEMAKE = "TARGET_DIR=${D} STAGING_DIR=${STAGING_DIR_TARGET} DESTDIR=${D} PKG_CONFIG=${STAGING_BINDIR_NATIVE}/pkg-config"
+EXTRA_OEMAKE = "OUT_DIR=${B} TARGET_DIR=${D} STAGING_DIR=${STAGING_DIR_TARGET} DESTDIR=${D} PKG_CONFIG=${STAGING_BINDIR_NATIVE}/pkg-config"
 
 do_compile(){
-    oe_runmake -C ${S} ${EXTRA_OEMAKE} all
+    oe_runmake -C ${S} ${EXTRA_OEMAKE} all 
 }
 
 do_install() {
-    install -d -m 0755 ${D}${libdir}
     install -d -m 0755 ${D}${includedir}
+    install -d -m 0755 ${D}${libdir}
 
-    install -D -m 0755 ${S}/*.so ${D}${libdir}/
-    install -D -m 0755 ${S}/inc/*.h ${D}${includedir}/
-
-    rm ${S}/*.so
+    install -D -m 0644 ${S}/inc/*.h ${D}${includedir}/
+    install -D -m 0644 ${B}/*.so ${D}${libdir}/
 }
 
 FILES_${PN} = " ${libdir}/* /etc/* "
diff --git a/recipes-multimedia/gst-aml-utils/gst-aml-dma-allocator.bb b/recipes-multimedia/gst-aml-utils/gst-aml-dma-allocator.bb
index d5ca9f6c..54332d30 100755
--- a/recipes-multimedia/gst-aml-utils/gst-aml-dma-allocator.bb
+++ b/recipes-multimedia/gst-aml-utils/gst-aml-dma-allocator.bb
@@ -13,21 +13,20 @@ SRCREV ?="${AUTOREV}"
 
 do_configure[noexec] = "1"
 
-EXTRA_OEMAKE = "TARGET_DIR=${D} STAGING_DIR=${STAGING_DIR_TARGET} DESTDIR=${D} PKG_CONFIG=${STAGING_BINDIR_NATIVE}/pkg-config"
+EXTRA_OEMAKE = "OUT_DIR=${B} TARGET_DIR=${D} STAGING_DIR=${STAGING_DIR_TARGET} DESTDIR=${D} PKG_CONFIG=${STAGING_BINDIR_NATIVE}/pkg-config"
 
 do_compile(){
     oe_runmake -C ${S} ${EXTRA_OEMAKE} all
 }
 
 do_install() {
+    install -d -m 0755 ${D}${includedir}
     install -d -m 0755 ${D}${libdir}
+
     install -d -m 0755 ${D}${libdir}/gstreamer-1.0/
-    install -d -m 0755 ${D}${includedir}
-    install -D -m 0755 ${S}/*.so ${D}${libdir}/
     mkdir -p ${D}${includedir}/gst/allocators/
-    install -D -m 0755 ${S}/inc/*.h ${D}${includedir}/gst/allocators/
-
-    rm ${S}/*.so
+    install -D -m 0644 ${S}/inc/*.h ${D}${includedir}/gst/allocators/
+    install -D -m 0644 ${B}/*.so ${D}${libdir}/
 }
 
 FILES_${PN} = " ${libdir}/* "
diff --git a/recipes-multimedia/gst-aml-utils/gst-aml-gfx2d.bb b/recipes-multimedia/gst-aml-utils/gst-aml-gfx2d.bb
index e41053a7..ce78db1d 100755
--- a/recipes-multimedia/gst-aml-utils/gst-aml-gfx2d.bb
+++ b/recipes-multimedia/gst-aml-utils/gst-aml-gfx2d.bb
@@ -13,21 +13,18 @@ SRCREV ?="${AUTOREV}"
 
 do_configure[noexec] = "1"
 
-EXTRA_OEMAKE = "TARGET_DIR=${D} STAGING_DIR=${STAGING_DIR_TARGET} DESTDIR=${D} PKG_CONFIG=${STAGING_BINDIR_NATIVE}/pkg-config"
+EXTRA_OEMAKE = "OUT_DIR=${B} TARGET_DIR=${D} STAGING_DIR=${STAGING_DIR_TARGET} DESTDIR=${D} PKG_CONFIG=${STAGING_BINDIR_NATIVE}/pkg-config"
 
 do_compile(){
     oe_runmake -C ${S} ${EXTRA_OEMAKE} all
 }
 
 do_install() {
-    install -d -m 0755 ${D}${libdir}
-    install -d -m 0755 ${D}${libdir}/gstreamer-1.0/
     install -d -m 0755 ${D}${includedir}
+    install -d -m 0755 ${D}${libdir}
 
-    install -D -m 0755 ${S}/*.so ${D}${libdir}/
-    install -D -m 0755 ${S}/inc/*.h ${D}${includedir}/
-
-    rm ${S}/*.so
+    install -D -m 0644 ${S}/inc/*.h ${D}${includedir}/
+    install -D -m 0644 ${B}/*.so ${D}${libdir}/
 }
 
 FILES_${PN} = " ${libdir}/* "
diff --git a/recipes-multimedia/gst-aml-utils/gst-jpeg-swdec.bb b/recipes-multimedia/gst-aml-utils/gst-jpeg-swdec.bb
index 034bdb99..d9ab2cc6 100755
--- a/recipes-multimedia/gst-aml-utils/gst-jpeg-swdec.bb
+++ b/recipes-multimedia/gst-aml-utils/gst-jpeg-swdec.bb
@@ -12,20 +12,18 @@ SRCREV ?="${AUTOREV}"
 
 do_configure[noexec] = "1"
 
-EXTRA_OEMAKE = "TARGET_DIR=${D} STAGING_DIR=${STAGING_DIR_TARGET} DESTDIR=${D} PKG_CONFIG=${STAGING_BINDIR_NATIVE}/pkg-config"
+EXTRA_OEMAKE = "OUT_DIR=${B} TARGET_DIR=${D} STAGING_DIR=${STAGING_DIR_TARGET} DESTDIR=${D} PKG_CONFIG=${STAGING_BINDIR_NATIVE}/pkg-config"
 
 do_compile(){
     oe_runmake -C ${S} ${EXTRA_OEMAKE} all
 }
 
 do_install() {
-    install -d -m 0755 ${D}${libdir}
-    install -d -m 0755 ${D}${libdir}/gstreamer-1.0/
     install -d -m 0755 ${D}${includedir}
-    install -D -m 0755 ${S}/*.so ${D}${libdir}/
-    install -D -m 0755 ${S}/inc/*.h ${D}${includedir}/
+    install -d -m 0755 ${D}${libdir}
 
-    rm ${S}/*.so
+    install -D -m 0644 ${S}/inc/*.h ${D}${includedir}/
+    install -D -m 0644 ${B}/*.so ${D}${libdir}/
 }
 
 FILES_${PN} = " ${libdir}/* "
diff --git a/recipes-multimedia/gst-aml-utils/gst-plugin-conv.bb b/recipes-multimedia/gst-aml-utils/gst-plugin-conv.bb
index c6c87d78..f1787f79 100644
--- a/recipes-multimedia/gst-aml-utils/gst-plugin-conv.bb
+++ b/recipes-multimedia/gst-aml-utils/gst-plugin-conv.bb
@@ -5,20 +5,19 @@ LIC_FILES_CHKSUM = "file://${COREBASE}/../meta-meson/license/AMLOGIC;md5=6c70138
 
 DEPENDS = " gstreamer1.0 gstreamer1.0-plugins-base"
 DEPENDS += " gst-aml-dma-allocator gst-aml-gfx2d"
-RDEPENDS_${PN} += " gst-aml-dma-allocator gst-aml-gfx2d libge2d"
+RDEPENDS_${PN} += " dma-allocator gst-aml-dma-allocator gst-aml-gfx2d libge2d"
 
 #For common patches
 SRC_URI_append = " ${@get_patch_list_with_path('${COREBASE}/../aml-patches/multimedia/gst-plugin-conv/', '../')}"
 
 SRCREV ?= "${AUTOREV}"
 
-S = "${WORKDIR}/git/"
-
+#S = "${WORKDIR}/git/"
 
 DEPENDS += "${@bb.utils.contains('EXTRA_OECONF', ' ', ' ', ' ', d)}"
 RDEPENDS_${PN} += "${@bb.utils.contains('EXTRA_OECONF', ' ', ' ', ' ', d)}"
 
-EXTRA_OEMAKE = "CROSS=${TARGET_PREFIX} TARGET_DIR=${STAGING_DIR_TARGET} STAGING_DIR=${D} DESTDIR=${D}"
+EXTRA_OEMAKE = "OUT_DIR=${B} TARGET_DIR=${D} STAGING_DIR=${STAGING_DIR_TARGET} DESTDIR=${D} CROSS=${TARGET_PREFIX}"
 
 inherit autotools pkgconfig features_check
 
diff --git a/recipes-multimedia/gst-aml-utils/gst-plugin-nn.bb b/recipes-multimedia/gst-aml-utils/gst-plugin-nn.bb
index 7e368831..504322d5 100644
--- a/recipes-multimedia/gst-aml-utils/gst-plugin-nn.bb
+++ b/recipes-multimedia/gst-aml-utils/gst-plugin-nn.bb
@@ -5,20 +5,19 @@ LIC_FILES_CHKSUM = "file://${COREBASE}/../meta-meson/license/AMLOGIC;md5=6c70138
 
 DEPENDS = " gstreamer1.0 gstreamer1.0-plugins-base"
 DEPENDS += " gst-aml-dma-allocator gst-aml-gfx2d detect-library"
-RDEPENDS_${PN} += " gst-aml-dma-allocator gst-aml-gfx2d detect-library npu-common libge2d"
+RDEPENDS_${PN} += " dma-allocator gst-aml-dma-allocator gst-aml-gfx2d detect-library npu-common libge2d"
 
 #For common patches
 SRC_URI_append = " ${@get_patch_list_with_path('${COREBASE}/../aml-patches/multimedia/gst-plugin-nn/', '../')}"
 
 SRCREV ?= "${AUTOREV}"
 
-S = "${WORKDIR}/git/"
-
+#S = "${WORKDIR}/git/"
 
 DEPENDS += "${@bb.utils.contains('EXTRA_OECONF', ' ', ' ', ' ', d)}"
 RDEPENDS_${PN} += "${@bb.utils.contains('EXTRA_OECONF', ' ', ' ', ' ', d)}"
 
-EXTRA_OEMAKE = "CROSS=${TARGET_PREFIX} TARGET_DIR=${D} STAGING_DIR=${STAGING_DIR_TARGET} DESTDIR=${D}"
+EXTRA_OEMAKE = "OUT_DIR=${B} TARGET_DIR=${D} STAGING_DIR=${STAGING_DIR_TARGET} DESTDIR=${D} CROSS=${TARGET_PREFIX}"
 
 inherit autotools pkgconfig features_check
 
diff --git a/recipes-multimedia/gst-aml-utils/gst-plugin-overlay.bb b/recipes-multimedia/gst-aml-utils/gst-plugin-overlay.bb
index fc34409a..3f2f1da1 100644
--- a/recipes-multimedia/gst-aml-utils/gst-plugin-overlay.bb
+++ b/recipes-multimedia/gst-aml-utils/gst-plugin-overlay.bb
@@ -5,20 +5,19 @@ LIC_FILES_CHKSUM = "file://${COREBASE}/../meta-meson/license/AMLOGIC;md5=6c70138
 
 DEPENDS = " gstreamer1.0 gstreamer1.0-plugins-base"
 DEPENDS += " gst-aml-dma-allocator gst-aml-gfx2d "
-RDEPENDS_${PN} += " gst-aml-dma-allocator gst-aml-gfx2d libge2d "
+RDEPENDS_${PN} += " dma-allocator gst-aml-dma-allocator gst-aml-gfx2d libge2d "
 
 #For common patches
 SRC_URI_append = " ${@get_patch_list_with_path('${COREBASE}/../aml-patches/multimedia/gst-plugin-overlay/', '../')}"
 
 SRCREV ?= "${AUTOREV}"
 
-S = "${WORKDIR}/git/"
-
+#S = "${WORKDIR}/git/"
 
 DEPENDS += "${@bb.utils.contains('EXTRA_OECONF', ' ', ' ', ' ', d)}"
 RDEPENDS_${PN} += "${@bb.utils.contains('EXTRA_OECONF', ' ', ' ', ' ', d)}"
 
-EXTRA_OEMAKE = "CROSS=${TARGET_PREFIX} TARGET_DIR=${D} STAGING_DIR=${STAGING_DIR_TARGET} DESTDIR=${D}"
+EXTRA_OEMAKE = "OUT_DIR=${B} TARGET_DIR=${D} STAGING_DIR=${STAGING_DIR_TARGET} DESTDIR=${D} CROSS=${TARGET_PREFIX}"
 
 inherit autotools pkgconfig features_check
 
-- 
2.25.1

